mm2src/coins/tendermint/tendermint_tx_history_v2.rs

```diff
@@ -0,0 +1,803 @@
+                coin: coin.platform_ticker().to_string(),
+                amount: big_decimal_from_sat_unsigned(fee_amount, coin.decimals()),
+                gas_limit: fee.gas_limit.value(),
+            })
+        }
+
+        struct TransferDetails {
+            from: String,
+            to: String,
+            denom: String,
+            amount: u64,
+        }
+
+        fn parse_transfer_values_from_events(tx_events: Vec<&Event>) -> Option<TransferDetails> {
+            let mut from: Option<String> = None;
+            let mut to: Option<String> = None;
+            let mut denom: Option<String> = None;
+            let mut amount: Option<u64> = None;
+
+            for event in tx_events {
+                if amount.is_none() && denom.is_none() {
+                    let amount_with_denom = event
+                        .attributes
+                        .iter()
+                        .find(|tag| tag.key.to_string() == AMOUNT_TAG_KEY)?
+                        .value
+                        .to_string();
+
+                    let extracted_amount: String = amount_with_denom.chars().take_while(|c| c.is_numeric()).collect();
+
+                    denom = Some(amount_with_denom.split(&extracted_amount).collect());
+                    amount = Some(extracted_amount.parse().ok()?);
+                }
+
+                match event.type_str.as_str() {
+                    COIN_SPENT_EVENT if from.is_none() => {
+                        from = Some(
+                            event
+                                .attributes
+                                .iter()
+                                .find(|tag| tag.key.to_string() == SPENDER_TAG_KEY)?
+                                .value
+                                .to_string(),
+                        );
+                    },
+                    COIN_RECEIVED_EVENT if to.is_none() => {
+                        to = Some(
+                            event
+                                .attributes
+                                .iter()
+                                .find(|tag| tag.key.to_string() == RECEIVER_TAG_KEY)?
+                                .value
+                                .to_string(),
+                        );
+                    },
+                    _ => {},
+                }
+            }
+
+            Some(TransferDetails {
+                from: from?,
+                to: to?,
+                denom: denom?,
+                amount: amount?,
+            })
+        }
+
+        fn get_transfer_details(tx_events: Vec<Event>, fee_amount_with_denom: String) -> Option<TransferDetails> {
+            // Filter out irrelevant events
+            let mut events: Vec<&Event> = tx_events
+                .iter()
+                .filter(|event| [COIN_SPENT_EVENT, COIN_RECEIVED_EVENT].contains(&event.type_str.as_str()))
+                .collect();
+
+            if events.len() > DEFAULT_TRANSFER_EVENT_COUNT {
```

Comment:
Will this work correctly if multiple assets are transferred in a single tx?

Reply:
I believe it will not work, I need to check it.